posthog/hogql_queries/web_analytics/stats_table.py

```diff
@@ -309,6 +312,66 @@ def to_path_bounce_query(self) -> ast.SelectQuery:
+        with self.timings.measure("frustration_metrics_query"):
+            # Base selects, always returns the breakdown value, and the total number of visitors
+            selects = [
+                ast.Alias(alias="context.columns.breakdown_value", expr=self._processed_breakdown_value()),
+                self._period_comparison_tuple("rage_clicks_count", "context.columns.rage_clicks", "sum"),
+                self._period_comparison_tuple("dead_clicks_count", "context.columns.dead_clicks", "sum"),
+                self._period_comparison_tuple("errors_count", "context.columns.errors", "sum"),
+            ]
+
+            query = ast.SelectQuery(
+                select=selects,
+                select_from=ast.JoinExpr(table=self._frustration_metrics_inner_query()),
+                group_by=[ast.Field(chain=["context.columns.breakdown_value"])],
+                order_by=self._frustration_metrics_order_by(columns=[select.alias for select in selects]),
+            )
+
+        return query
+
+    def _frustration_metrics_inner_query(self):
+        query = parse_select(
+            """
+            SELECT
+                any(person_id) AS filtered_person_id,
+                count() AS filtered_pageview_count,
+                {breakdown_value} AS breakdown_value,
+                countIf(events.event = '$rageclick') AS rage_clicks_count,
+                countIf(events.event = '$dead_click') AS dead_clicks_count,
+                countIf(events.event = '$exception') AS errors_count,
+                session.session_id AS session_id,
+                min(session.$start_timestamp) as start_timestamp
+            FROM events
+            WHERE and({inside_periods}, {event_where}, {all_properties}, {where_breakdown})
+            GROUP BY session_id, breakdown_value
+            """,
+            timings=self.timings,
+            placeholders={
+                "breakdown_value": self._counts_breakdown_value(),
+                "event_where": parse_expr("events.event IN ('$pageview', '$rageclick', '$dead_click', '$exception')"),
+                "all_properties": self._all_properties(),
+                "where_breakdown": self.where_breakdown(),
+                "inside_periods": self._periods_expression(),
+            },
+        )
+
+        assert isinstance(query, ast.SelectQuery)
+        return query
+
+    def _frustration_metrics_order_by(self, columns: list[str]) -> list[ast.OrderExpr] | None:
+        return [
+            expr
+            for expr in [
+                # A very simplistic "frustration score" while we don't fancy doing something more sophisticated
+                ast.OrderExpr(expr=ast.Field(chain=["context.columns.rage_clicks"]), order="DESC"),
+                ast.OrderExpr(expr=ast.Field(chain=["context.columns.dead_clicks"]), order="DESC"),
+                ast.OrderExpr(expr=ast.Field(chain=["context.columns.errors"]), order="DESC"),
+            ]
+            if expr is not None
+        ]
```

Comment:
This `if expr is not None` check is not doing anything. We're only building the AST here, so you're not comparing the actual numbers or anything.